application/common/models/Method.php

```diff
@@ -61,4 +93,156 @@ public function getRawPhoneNumber()
+            }
+        } else {
+            throw new \Exception('Invalid method type', 1461432437);
+        }
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function sendVerificationEmail()
+    {
+        Verification::sendEmail(
+            $this->value,
+            'Verification required - New account recovery method added',
+            '@common/mail/method/verify',
+            $this->verification_code,
+            $this->user,
+            null,
+            $this->user->getId(),
+            'New email method',
+            'A new email method has been added and verification sent to ' . $this->getMaskedValue()
+        );
+    }
+
+    /**
+     * @return string
+     * @throws \Exception
+     */
+    public function sendVerificationPhone()
+    {
+        return Verification::sendPhone(
+            $this->getRawPhoneNumber(),
+            $this->verification_code,
+            $this->user->getId(),
+            'New phone method',
+            'A new phone method has been added and verification sent to ' . $this->getMaskedValue()
+        );
+    }
+
+    /**
+     * @param string $userSubmitted
+     * @return bool
+     */
+    public function isUserProvidedCodeCorrect($userSubmitted)
+    {
+        if ($this->type === self::TYPE_EMAIL) {
+            return Verification::isEmailCodeValid($this->verification_code, $userSubmitted);
+        } elseif ($this->type === self::TYPE_PHONE) {
+            return Verification::isPhoneCodeValid($this->verification_code, $userSubmitted);
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Validate user submitted code and update record to be verified if valid
+     * @param string $userSubmitted
+     * @throws \Exception
+     */
+    public function validateAndSetAsVerified($userSubmitted)
+    {
+        if ( ! $this->isUserProvidedCodeCorrect($userSubmitted)) {
+            $this->verification_attempts++;
+            $this->save();
+            throw new \Exception('Invalid verification code', 1461442988);
+        }
+
+        /*
+         * Update attributes to be verified
+         */
+        $this->verification_code = null;
+        $this->verification_expires = null;
+        $this->verification_attempts = null;
+        $this->verified = 1;
+
+        if ( ! $this->save()) {
+            throw new \Exception('Unable to set method as verified', 1461442990);
+        }
+    }
+
+    /**
+     * Delete all method records that are not verified and verification_expires date is in the past
+     * @throws \Exception
+     */
+    public static function deleteExpiredUnverifiedMethods()
+    {
+        $methods = Method::find()->where(['verified' => 0])
+                                 ->andWhere(['<', 'verification_expires', Utils::getDatetime()])
+                                 ->all();
+
+        foreach ($methods as $method) {
```

Comment:
True, but I specifically want to use the model's delete method so it can do anything else needed like deleting foreign key records and such.